src/main/java/com/example/bookstoreapp/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.example.bookstoreapp.service.impl;
+
+import com.example.bookstoreapp.dto.user.UserRegistrationRequestDto;
+import com.example.bookstoreapp.dto.user.UserResponseDto;
+import com.example.bookstoreapp.exception.RegistrationException;
+import com.example.bookstoreapp.mapper.UserMapper;
+import com.example.bookstoreapp.model.RoleName;
+import com.example.bookstoreapp.model.User;
+import com.example.bookstoreapp.repository.book.role.RoleRepository;
+import com.example.bookstoreapp.repository.book.user.UserRepository;
+import com.example.bookstoreapp.service.UserService;
+import java.util.Arrays;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+@Component
```

Comment:
```suggestion
@Service 
```
It's no difference technically  but if your class has business  logic and is named ...Service use Service annotation